app/controllers/vm_common.rb

```diff
@@ -1370,7 +1370,7 @@ def console_before_task(console_type)
   # Task complete, show error or launch console using VNC/MKS/VMRC task info
   def console_after_task(console_type)
     miq_task = MiqTask.find(params[:task_id])
-    if miq_task.status == "Error" || miq_task.task_results.blank?
+    unless miq_task.results_ready?
```

Comment:
Is there a non-Error status they can be in besides Ok?  I don't remember if there's a running...

Can you add tests for the results_ready? method based on how all of these controller conditions currently work?

For example:
"task status is Error or task_results is blank"
"task status is NOT Ok"
"task_results are blank or task status is NOT ok"
"task_results are blank"